### PR TITLE
Implement Social Media Monitoring and Alert System Functionality in Cookbook

### PR DESCRIPTION
**Pull request by Mefisto04**
### PR Description:
This pull request improves the functionality of the Social Media Monitoring and Alert System by refining the trend analysis and alert sending mechanisms. The code has been modified to ensure that trends are extracted correctly and formatted appropriately before being passed to the alerting function. 

### Key Changes:
- Updated the `analyze_trends` function to return a list of trending posts.
- Adjusted the `send_alert` function to accept the correct input format and improved error handling.

### Example Function Snippet:
```python
def analyze_trends(posts, threshold):
    trends = [post for post in posts if posts.count(post) >= threshold]
    return {"trends": trends}  # Returns a list of trending posts

def send_alert(trends):
    if not trends:
        return {"status": "No trends detected to alert."}
    # Additional logic to send alerts...
```

----

> [!IMPORTANT]
> Adds social media monitoring and alert system with trend analysis and alert functions in `04_social_media_monitoring_and_alert_system.py`.
> 
>   - **New File**:
>     - `04_social_media_monitoring_and_alert_system.py` added for social media monitoring and alert system.
>   - **Functions**:
>     - `analyze_trends()` updated to return a list of trending posts based on a threshold.
>     - `send_alert()` updated to handle empty trends and send alerts using `SEND_ALERT_TASK_UUID`.
>   - **Tasks**:
>     - Defines `monitor_social_media_task`, `analyze_trends_task`, and `send_alert_task` using YAML schemas for task creation and execution.
>   - **Misc**:
>     - Adds `print_output()` to display monitoring, trends, and alert results.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=julep-ai%2Fjulep&utm_source=github&utm_medium=referral)<sup> for 3a606b417f32c18f44e6799638512ce223171309. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by Mefisto04** [comment]
@creatorrr i have implemented the 04 codebook mention in the issue of Batch1, please review and merge this.

**Comment by Mefisto04** [comment]
Hey @Vedantsahai18 @ayush2390 please review this pr

**Comment by Vedantsahai18** [comment]
could please remove the notebook no as it might lead to a conflict

**Comment by Mefisto04** [comment]
@Vedantsahai18 done, i have changed the file and and remove the number from it.

File: cookbooks/04_social_media_monitoring_and_alert_system.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    print("Trends Analyzed:")
+    print(f"Trending posts based on the defined threshold:\n{', '.join(trends_result['trends'])}\n")
+
+    print("Alert Status:")
+    print(f"Alert status: {alert_result.get('status', 'No alert sent')}\n")
+
+keywords = ["AI", "Healthcare", "Data"]
+threshold = 2
+
+monitor_result = monitor_social_media(keywords)
```
[the remark this answers is not in the document]
Comment (Vedantsahai18) [reply cookbooks/04_social_media_monitoring_and_alert_system.py]: please have a look into this!

**Comment by Vedantsahai18** [comment]
In notebook/py script `monitor_social_media` is not defined. This will cause a NameError.

**Comment by Mefisto04** [comment]
hey @Vedantsahai18 , i have changed the code, please review once and merge it, it contain all the functionality and working fine

**Pull request closed without merging**